src/services/severa-api-service.ts

```diff
@@ -33,200 +34,113 @@ export const CreateSeveraApiService = (): SeveraApiService => {
-              "Client_Id": process.env.SEVERA_DEMO_CLIENT_ID,
-              "Content-Type": "application/json",
-            },
-          });
-    
-          if (!allKeywordsResponse.ok) {
-            throw new Error(`Failed to fetch all keywords: ${allKeywordsResponse.status} - ${allKeywordsResponse.statusText}`);
-          }
-    
-          const allKeywords = await allKeywordsResponse.json();
-          let keywordToAdd = allKeywords.find((kw: { keyword: string }) => kw.keyword === "isSeveraOptIn");
-
-          if (!keywordToAdd) {
-
-            const createKeywordUrl = `${baseUrl}/v1/keywords`;
-            const createKeywordResponse = await fetch(createKeywordUrl, {
-              method: "POST",
-              headers: {
-                Authorization: `Bearer ${await getSeveraAccessToken()}`,
-                "Client_Id": process.env.SEVERA_DEMO_CLIENT_ID,
-                "Content-Type": "application/json",
-              },
-              body: JSON.stringify({
-                keyword: "isSeveraOptIn",
-                category: "User",
-                isActive: true,
-              }),
-            });
-    
-            if (!createKeywordResponse.ok) {
-              throw new Error(`Failed to create 'isSeveraOptIn' keyword: ${createKeywordResponse.status} - ${createKeywordResponse.statusText}`);
-            }
-    
-            keywordToAdd = await createKeywordResponse.json();
-          }
-
-          if (keywordToAdd) {
-            const updateKeywordUrl = `${baseUrl}/v1/users/${user.guid}/keywords/${keywordToAdd.guid}`;
-            const updateResponse = await fetch(updateKeywordUrl, {
-              method: "POST",
-              headers: {
-                Authorization: `Bearer ${await getSeveraAccessToken()}`,
-                "Client_Id": process.env.SEVERA_DEMO_CLIENT_ID,
-                "Content-Type": "application/json",
-              },
-              body: JSON.stringify({
-                value: isSeveraOptIn,
-              }),
-            });
-    
-            if (!updateResponse.ok) {
-              throw new Error(`Failed to update Severa keyword: ${updateResponse.status} - ${updateResponse.statusText}`);
-            }
-          }
+
+        const keywords = await getUserKeywords(user.guid);
+        const existingKeywordForUser = keywords.find(
+          (kw: { keyword: string }) => kw.keyword === "isSeveraOptIn"
+        );
+
+        if (!existingKeywordForUser) {
+          const updatedKeyword = await updateSeveraOptInKeyword(
+            user.guid,
+            isSeveraOptIn,
+            isSeveraOptInKeyword.guid
+          );
+          return {
+            guid: user.guid,
+            isSeveraOptIn: updatedKeyword.value,
+            email: testEmail
+          };
         }
         return {
           guid: user.guid,
-          isSeveraOptIn, 
-          email,
+          isSeveraOptIn: existingKeywordForUser.value,
+          email: testEmail,
+          testUserId: testUserId
```

Comment:
Why is this testUserId needed here, at this point won't it be the same as user.guid?

Reply:
You are returning from here though, which means you can return these in the body of the handler return- doesn't that make more sense?